### PR TITLE
fix(watch): check child message data to be an object

### PR DESCRIPTION
**Pull request by pedablio**

### CODE REVIEW

**Comment by privatenumber** [comment]
Is there a specific problem you're addressing?

**Comment by pedablio** [comment]
I'm having problem using watch mode
It seems that when receiving the dependencies by
`runProcess.on('message', (data) => {`
The variable data also receives a string with the text 'ready' in my case
Giving the error: **Cannot use 'in' operator to search for 'type' in ready**

**Comment by endel** [comment]
I'm also getting this error when running `tsx watch src/index.ts`

```
Cannot use 'in' operator to search for 'type' in ready
```

The piece of code that seems to be related to this issue come from these files:

```
node_modules/chalk/node_modules/supports-color/index.js
84:		if (['TRAVIS', 'CIRCLECI', 'APPVEYOR', 'GITLAB_CI', 'GITHUB_ACTIONS', 'BUILDKITE'].some(sign => sign in env) || env.CI_NAME === 'codeship') {

node_modules/supports-color/index.js
97:		if (['TRAVIS', 'CIRCLECI', 'APPVEYOR', 'GITLAB_CI', 'GITHUB_ACTIONS', 'BUILDKITE', 'DRONE'].some(sign => sign in env) || env.CI_NAME === 'codeship') {
```

File: src/watch/utils.ts [hunk 1]
```diff
@@ -35,6 +35,7 @@ export function isDependencyPath(
 ): data is { type: 'dependency'; path: string } {
 	return (
 		data
+		&& typeof data === 'object'
 		&& 'type' in data
```
Comment (privatenumber) [inline src/watch/utils.ts:39]: Can you remove this line?

**Pull request merged**